### DISCUSSION TITLE
Error 500 validating a form field of type datetimeImmutable with LiveComponent

### DISCUSSION THREAD

**Discussion by tallakoundoul** [symfony/symfony#59001] [Q&A]
Here is my code:
Entity vehicule:
`#[ORM\Entity(repositoryClass: VehiculeRepository::class)]
class Vehicule
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[NotBlank(message: "Matricule invalide !")]
    private ?string $matricule = null;

    #[ORM\Column(length: 255)]
    #[NotBlank(message: "Numéro de chassis invalide !")]
    private ?string $numero_chassis = null;

    #[ORM\Column(nullable: true)]
    private ?int $compteurActuel = null;

    #[ORM\Column]
    private ?\DateTimeImmutable $dateTraitement = null;

    #[ORM\Column]
    private ?bool $actif = null;

    #[NotBlank(message: "La catégorie est requise.")]
    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    #[ORM\JoinColumn(nullable: false)]
    private ?CategorieVehicule $categorieVehicule = null;

    #[NotBlank(message: "Le type de carburant est requis.")]
    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    #[ORM\JoinColumn(nullable: false)]
    private ?TypeCarburant $typeCarburant = null;

    #[NotBlank(message: "La couleur est requise.")]
    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Couleur $couleur = null;

    /**
     * @var Collection<int, Location>
     */
    #[ORM\ManyToMany(targetEntity: Location::class, mappedBy: 'vehicule')]
    private Collection $locations;

    #[NotBlank(message: "Le compteur initial est requis.")]
    #[ORM\Column]
    private ?int $compteurInitial = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $commentaire = null;

    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    #[ORM\JoinColumn(nullable: true)]
    private ?EntrepriseProprietaire $entrepriseProprietaire = null;

    #[NotBlank(message: "La marque est requise.")]
    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    private ?Marque $marque = null;

    #[NotBlank(message: "Le modele est requis.")]
    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    private ?Modele $modele = null;

    #[NotBlank(message: "Cette date est requise.")]
    #[ORM\Column(type: Types::DATE_IMMUTABLE)]
    private ?\DateTimeImmutable $dateAcquisition = null;

    #[NotBlank(message: "Cette date est requise.")]
    #[ORM\Column(type: Types::DATE_IMMUTABLE)]
    private ?\DateTimeImmutable $dateMiseEnCirculation = null;

    #[NotBlank(message:"La puissance est requise !")]
    #[ORM\Column]
    private ?int $puissance = null;

    #[ORM\ManyToOne(inversedBy: 'vehicules')]
    private ?Tarification $tarification = null;

    public function __construct()
    {
        $this->dateTraitement = new DateTimeImmutable();
        $this->locations = new ArrayCollection();
    }

... Getters And Setters
}

VehicculeType:

class VehiculeType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder = new DynamicFormBuilder($builder);

        $builder
            ->add('matricule', TextType::class, [
                'label' => 'Matricule',
                'attr' => [
                    'placeholder' => 'matricule',
                ],
            ])
            ->add('numero_chassis', TextType::class, [
                'label' => 'Numéro de chassis',
                'attr' => [
                    'placeholder' => 'Numéro de chassis',
                ],
            ])
            ->add('actif', CheckboxType::class, [
                'label_attr' => [
                    'class' => 'checkbox-switch',
                ]
            ])
            ->add('compteurInitial', IntegerType::class, [
                'label' => 'Compteur Initial',
                'attr' => [
                    'placeholder' => 'Compteur Initial',
                ],
            ])
            ->add('puissance', IntegerType::class, [
                
            ])
            ->add('dateAcquisition', DateType::class, [
                'widget' => 'single_text',
            ])
            ->add('dateMiseEnCirculation', DateType::class, [
                'widget' => 'single_text',
            ])
            ->add('categorieVehicule', EntityType::class, [
                'class' => CategorieVehicule::class,
                'choice_label' => fn (CategorieVehicule $c): string => $c->getNom(),
                'placeholder' => '- Choisir une catégorie -',
                'required' => false
            ])
            ->addDependent('marque' , 'categorieVehicule', 
                function (DependentField $field, ?CategorieVehicule $categorie) {
                $field->add(EntityType::class, [
                    'class' => Marque::class,
                    'placeholder' => null === $categorie ? 'Choisissez une catégorie d\'abord' : '',
                    'choices' => $categorie?->getMarques(),
                    'choice_label' => fn (Marque $m): string => $m->getNom(),
                    'disabled' => null === $categorie,
                    'autocomplete' => true
                ]);
            })
            ->addDependent('modele' , 'marque', 
                function (DependentField $field, ?Marque $marque) {
                $field->add(EntityType::class, [
                    'class' => Modele::class,
                    'placeholder' => null === $marque ? 'Choisissez un modele d\'abord' : '',
                    'choices' => $marque?->getModeles(),
                    'choice_label' => fn (Modele $m): string => $m->getNom(),
                    'disabled' => null === $marque,
                    'autocomplete' => true
                ]);
            })
            ->add('typeCarburant', EntityType::class, [
                'class' => TypeCarburant::class,
                'choice_label' => fn (TypeCarburant $tc): string => $tc->getNom(),
            ])
            ->add('couleur', EntityType::class, [
                'class' => Couleur::class,
                'choice_label' => fn (Couleur $c): string => $c->getNom(),
            ])
            ->add('tarification', EntityType::class, [
                'class' => Tarification::class,
                'choice_label' => fn (Tarification $t): string => $t->getNom(),
                'placeholder' => '- Choisir une catégorie -',
                'required' => false
            ])
            ->add('commentaire', TextType::class, [
                'label' => 'Commentaire',
                'attr' => [
                    'placeholder' => 'Commentaire',
                ],
            ])
            ->add('Enregistrer', SubmitType::class)
            ;
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'data_class' => Vehicule::class,
        ]);
    }
}

The controller:
#[Route('/vehicules/new', name: 'new_vehicule', methods: ['POST', 'GET'])]
    public function new(Request $request): Response
    {
        $vehicule = new Vehicule();
        $form = $this->createForm(VehiculeType::class, $vehicule);
        $form->handleRequest($request);

        // if ($form->isSubmitted() && $form->isValid()) {
        //     $vehicule = $form->getData();
        //     $vehicule->setDateTraitement(new DateTimeImmutable());
        //     $em->persist($vehicule);
        //     $em->flush();

        //     $this->addFlash("success", "Client ajouté avec succes");

        //     return $this->redirectToRoute('list_vehicule');
        // }

        return $this->render('vehicule/new.html.twig', [
            'form' => $form
        ]);
    }

VehiculeF Form component:

#[AsLiveComponent]
final class VehiculeForm extends AbstractController
{
    use DefaultActionTrait;
    use ComponentWithFormTrait;

    #[LiveProp(fieldName: 'data')]
    public ?Vehicule $vehicule = null;

    protected function instantiateForm(): FormInterface
    {
        return $this->createForm(VehiculeType::class, $this->vehicule);
    }

    #[LiveAction]
    public function save(EntityManagerInterface $em) {
        $this->submitForm();

        $form = $this->getForm();

        // if ($form->isSubmitted() && $form->isValid()) {

            /** @var Vehicule $vehicule */
            $vehicule = $form->getData();
            $vehicule->setDateTraitement(new DateTimeImmutable());

            $em->persist($vehicule);
            $em->flush();

            $this->addFlash("success", "Client ajouté avec succes");

            return $this->redirectToRoute('list_vehicule');
        // }
    }
}

the template: templates\components\VehiculeForm.html.twig
<div{{ attributes }}>
    {{ form_start(form, {
        attr: {
            'data-action': 'live#action',
            'data-live-action-param': 'save'
        }
    }) }}
        <div class="row my-4">
            {{ form_row(form.matricule) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.numero_chassis) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.actif) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.compteurInitial) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.puissance) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.dateAcquisition) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.dateMiseEnCirculation) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.categorieVehicule) }}
        </div>
        <div class="row my-4">
            {% if form.marque is defined %}
                {{ form_row(form.marque) }}
            {% endif %}
        </div>
        <div class="row my-4">
            {% if form.modele is defined %}
                {{ form_row(form.modele) }}
            {% endif %}
        </div>
        <div class="row my-4">
            {{ form_row(form.typeCarburant) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.couleur) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.tarification) }}
        </div>
        <div class="row my-4">
            {{ form_row(form.commentaire) }}
        </div>
    {{ form_end(form) }}
</div>

new.html.twig :

{% extends 'base.html.twig' %}

{% block body %}
{# <twig:VehiculeForm /> #}
{{ component('VehiculeForm', {
    form: form
}) }}
{% endblock %}

`

When I choose one of the two dates on the form, a 500 error is sent by the server via Ajax.
Example by entering the date: "dateAcquisition", I get the following error:
Expected argument of type "DateTimeImmutable", "null" given at property path "dateMiseEnCirculation".

![error_live_component](https://github.com/user-attachments/assets/292ea4e6-2c5f-4e93-a744-6ad07782d92f)

Please help me resolve this issue.
I'm using synfony 7.1 with PHP 8.2
